clar2wasm/src/standard/standard.wat

```diff
@@ -1468,6 +1465,40 @@
         (i32.store offset=28 (local.get $origin) (i32.add (i32.load offset=28 (local.get $origin)) (local.get $h)))
     )
 
+    ;;
+    ;; logical not implementation
+    ;;
+    (type $not-func (func (param $bool_in i32) (result i32)))
+    (func $not (type $not-func) (param $bool_in i32) (result i32)
+        (i32.eqz (local.get $bool_in))
+    )
+    (export "not" (func $not))
+
+    ;;
+    ;; xor implementation
+    ;;
+    (type $xor-func (func (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)))
+    (func $xor (type $xor-func) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)
+        (local.get $a_lo)
+        (local.get $a_hi)
+        (local.get $b_lo)
+        (local.get $b_hi)
+        (call $bit-xor)
+    )
+    (export "xor" (func $xor))
+
+    ;;
+    ;; is-eq implementation
+    ;;
+    (type $is-eq-func (func (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i32)))
+    (func $is-eq (type $is-eq-func) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i32)
+        (i32.eq
```

Comment:
This will return true if both the lo and the hi parts are unequal. I think instead you can `and` the results together.